src/app_model/registries/_keybindings_reg.py

```diff
@@ -26,12 +26,30 @@ class _RegisteredKeyBinding(NamedTuple):
 
 
 class KeyBindingsRegistry:
-    """Registry for keybindings."""
+    """Registry for keybindings.
+
+    Attributes
+    ----------
+    filter_keybinding : Callable[[KeyBinding], str] | None
+        Optional function for filtering out invalid `KeyBinding`s.
+        Callable should return a error message string if `KeyBinding` is invalid
+        or empty string if `KeyBinding` is valid.
+    """
 
     registered = Signal()
 
     def __init__(self) -> None:
         self._keybindings: list[_RegisteredKeyBinding] = []
+        self._filter_keybinding: Callable[[KeyBinding], str] | None = None
+
+    @property
+    def filter_keybinding(self) -> Callable[[KeyBinding], str] | None:
+        """Return the `filter_keybinding`."""
+        return self._filter_keybinding
+
+    @filter_keybinding.setter
+    def filter_keybinding(self, filter: Callable[[KeyBinding], str] | None) -> None:
+        self._filter_keybinding = filter
```

Comment:
Should we be doing any type checking?

Reply:
yes, should check that it's either None or callable, but don't need to verify parameter/return types